documentation/update-readme

SAleksiev opened the pull request:
- Update docs related to newly exposed method trackNotificationEvent

SAleksiev committed:
documentation/update-readme
- Update docs related to newly exposed method trackNotificationEvent

SAleksiev committed:
- remove automatically

SAleksiev committed:
- update docs params and examples

SAleksiev commented on README.md:
```diff
@@ -97,15 +107,32 @@ struct ContentView: View {
+
+*For full documentation related to react-native setup follow the link here*
+`TODO: - Add link when repo is deployed`
+
```
Once the react-native bridge is available, I will add the URL to the repository here.

ahkjeldsen commented on README.md:
````diff
@@ -62,6 +62,16 @@ public func trackSearchEvent(query: String, isSuccessful: Bool, numberOfResults:
+
+- Track push notification event
+```swift
+public func trackNotificationEvent(uid: String, s: String, k: String)
````
Can we use typed out names, like `state` instead of `s`. Also the `k` is optional I suppose. Is it possible to pass null or leave out the parameter?

SAleksiev replied:
Until now, all signatures of the methods have been in the same way as the open-api spec. Furthermore, we have well documented docs where each parameter is explained both on github as well as xcode project.

Example:

/// MARK: - Track push notification
/// - Parameters:
///   - uid: Unique identifier for the notification received
///   - s: The state of the notification - received or opened
///   - k: The key of the notification
///   example: trackNotification(uid: notification.request.identifier, s: "received", k: "some key")

In addition to that, we made all parameters for each method required and at the moment you cannot pass null or leave it. 

However, we can update the method signitures if thats wanted as well as enahnce paramteres name and pottentialy make any of them optional.

As I always like to say - your call.

ahkjeldsen replied:
The openAPI spec is made as miniscule as possible, just to save on bytes. The code calling the API does not need to be that. So for my sake, I'd prefer if the parameter names where more explicit.

I'd also prefer if the methods made available in the SDK would match the behavior from the openAPI spec, i.e. some parameters are optional.

I cant really make the call that you should spend time on changing this, if it's going to take a long time 😅

muskanstha replied:
I agree with making the parameter names more explicit

SAleksiev replied:
The docs are updated, please validate & approve. @muskanstha @ahkjeldsen 
We should not merge, until 1.2.0 is ready to be pushed.

muskanstha commented on README.md:
```diff
@@ -140,9 +163,34 @@ func refreshCart() {
 
-    - **k:** The event key describing the type of event ("cart.refresh" in this example).
-    - **a:** A dictionary of key-value pairs detailing attributes of the event. Here, we’re providing cart details like item count and value.
-    
-**Note:** When using the trackCustomEvent method, it is crucial to ensure the uniqueness of the keys in the attributes dictionary. Duplicate keys are not recommended. If duplication occurs, the first instance of the key-value pair will be prioritised, and subsequent duplicates will be ignored. Please take care to avoid key duplication to ensure accurate event tracking.
+**Note:** When using the trackNotificationEvent method, it is crucial to ensure the uniqueness of the uids passed as parameter.
```
An example for the key could be useful here to show what it could be used for.

muskanstha reviewed (approved):
Let's also align the use of bold and code for highlighing the parameters and function names.

SAleksiev committed:
documentation/update-readme
- Update docs with new method signatures
- Add missing react-native URL to the docs

SAleksiev committed:
Readme
- Update signature of methods as well as descriptions to keys of notification parameter

SAleksiev committed:
update-readme
- Add notification name as part of the docs

SAleksiev committed:
- bump version in package
